statistics/handle/cache/statscacheinner.go

```diff
@@ -26,64 +26,94 @@ import (
 	"github.com/pingcap/tidb/sessionctx"
 	"github.com/pingcap/tidb/sessionctx/variable"
 	"github.com/pingcap/tidb/statistics"
-	"github.com/pingcap/tidb/statistics/handle/internal/cache"
-	"github.com/pingcap/tidb/statistics/handle/internal/cache/lru"
-	"github.com/pingcap/tidb/statistics/handle/internal/cache/mapcache"
+	"github.com/pingcap/tidb/statistics/handle/cache/internal"
+	"github.com/pingcap/tidb/statistics/handle/cache/internal/lru"
+	"github.com/pingcap/tidb/statistics/handle/cache/internal/mapcache"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/util/chunk"
 	"github.com/pingcap/tidb/util/sqlexec"
 	"github.com/pingcap/tidb/util/syncutil"
 )
 
-func newStatsCache() statsCache {
+type TableStatsOption struct {
+	byQuery bool
+}
+
+func (t *TableStatsOption) ByQuery() bool {
+	return t.byQuery
+}
+
+// TableStatsOpt used to edit getTableStatsOption
+type TableStatsOpt func(*TableStatsOption)
+
+// WithTableStatsByQuery indicates user needed
+func WithTableStatsByQuery() TableStatsOpt {
+	return func(option *TableStatsOption) {
+		option.byQuery = true
+	}
+}
+
+func NewStatsCacheWrapper() StatsCacheWrapper {
 	enableQuota := config.GetGlobalConfig().Performance.EnableStatsCacheMemQuota
 	if enableQuota {
 		capacity := variable.StatsCacheMemQuota.Load()
-		return statsCache{
+		return StatsCacheWrapper{
 			StatsCacheInner: lru.NewStatsLruCache(capacity),
 		}
 	}
-	return statsCache{
+	return StatsCacheWrapper{
 		StatsCacheInner: mapcache.NewMapCache(),
 	}
 }
 
-// statsCache caches the tables in memory for Handle.
-type statsCache struct {
-	cache.StatsCacheInner
+// StatsCacheWrapper caches the tables in memory for Handle.
+type StatsCacheWrapper struct {
```

Comment:
I think all methods of `StatsCacheWrapper` should use the pointer receiver instead of the value receiver.